### PR TITLE
[WIP] Fix the oracle hardhat demo

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR fixes the oracle hardhat demo

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: packages/valory/services/oracle_hardhat/service.yaml [hunk 1]
```diff
@@ -14,23 +14,17 @@ extra:
...
       service_registry_address: '0x998abeb3E57409262aE5b751f60747921B33613E'
-      on_chain_service_id: 1
+      share_tm_config_on_startup: ${SHARE_TM_CONFIG_ON_STARTUP:bool:false}
+      on_chain_service_id: ${SERVICE_ID:int:1}
   benchmark_persistence_params:
-    args: &id003
-      log_dir: /benchmarks
-  valory_monitoring_backend:
     args: &id002
-      url: https://oracle-server.staging.autonolas.tech/deposit
```
Comment (angrybayblade) [inline packages/valory/services/oracle_hardhat/service.yaml:33]: Not required for demo

**Review by Adamantios — CHANGES_REQUESTED** [review]
I do not understand the purpose of this PR. The service configuration is correct and needs no changes, assuming that a user runs it using the registries image. This PR among other things removes the participants. How would the service work without participants?

We need the minimum number of changes here. Please close this PR and open a PR that only updates the docs to advise the user how to run the demo using the registries image.

File: packages/valory/services/oracle_hardhat/service.yaml [hunk 2]
```diff
@@ -15,22 +15,21 @@ extra:
...
       setup:
         all_participants:
-        - - '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266'
-          - '0x70997970C51812dc3A010C7d01b50e0d17dc79C8'
-          - '0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC'
-          - '0x90F79bf6EB2c4f870365E785982E1f101E93b906'
+        - - '0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65'
+          - '0x9965507D1a55bcC2695C58ba16FB37d819B0A4dc'
+          - '0x976EA74026E726554dB657fA54763abd0C3a0aa9'
+          - '0x14dC79964da2C08b23698B3D3cc7Ca32193d9955'
```
Comment (angrybayblade) [inline packages/valory/services/oracle_hardhat/service.yaml:21]: I've added the all participants list but the order is different

File: docs/index.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ Once you have {{set_up_system}} to work with the Open Autonomy framework, you ca
 4. Build the service deployment.
 
     ```bash
-    autonomy deploy build keys.json --aev
+    autonomy deploy build keys.json --aev -ltm
```
Comment (angrybayblade) [inline docs/index.md:67]: Required after the `v0.9.0` bump

File: packages/valory/services/oracle_hardhat/service.yaml [hunk 4]
```diff
@@ -113,17 +106,15 @@ type: skill
...
     benchmark_tool:
-      args: *id003
+      args: *id002
 ---
 public_id: valory/ledger:0.19.0
 type: connection
 config:
   ledger_apis:
     ethereum:
-      address: ${SERVICE_ORACLE_HARDHAT_RPC:str:http://hardhat:8545}
```
Comment (angrybayblade) [inline packages/valory/services/oracle_hardhat/service.yaml:126]: Can be only used with the dev mode

File: packages/valory/services/oracle_hardhat/service.yaml [hunk 5]
```diff
@@ -113,17 +106,15 @@ type: skill
...
+      address: ${SERVICE_ORACLE_HARDHAT_RPC:str:http://host.docker.internal:8545}
```
Comment (angrybayblade) [inline packages/valory/services/oracle_hardhat/service.yaml:117]: Will work after https://github.com/valory-xyz/open-autonomy/pull/1843

**Comment by angrybayblade** [comment]
> I do not understand the purpose of this PR. The service configuration is correct and needs no changes, assuming that a user runs it using the registries image. This PR among other things removes the participants. How would the service work without participants?
> 
> We need the minimum number of changes here. Please close this PR and open a PR that only updates the docs to advise the user how to run the demo using the registries image.

The service configuration is not correct, it contains the `http://hardhat:8545` as the ledger RPC which can be only used in the dev mode and we're not using the dev mode for the demo. The issue with the docker host gateway has been fixed on the open autonomy.

We don't need to use the registries image, the demo works fine with the default hardhat image.

**Review by Adamantios — COMMENTED** [review]
Looks good apart from https://github.com/valory-xyz/price-oracle/pull/49#discussion_r1114416943.

Let's wait for `0.9.1` release and see if these steps work.

File: packages/valory/services/oracle_hardhat/service.yaml (hunk 4 above)
Comment (Adamantios) [reply packages/valory/services/oracle_hardhat/service.yaml:126]: No can also be used without dev mode. Check #52.

File: packages/valory/services/oracle_hardhat/service.yaml (hunk 5 above)
Comment (Adamantios) [reply packages/valory/services/oracle_hardhat/service.yaml:117]: No, this did not work.

**Comment by Adamantios** [comment]
Closing this PR as it performs unnecessary changes. Nothing needs to change on the service side. Only the docs need to be updated for the demo to run. Replaced by #52.

**Pull request closed without merging**